### PR TITLE
fix(make): ensures projects version of operator-sdk is used

### LINKED ISSUE #694: Makefile does not use operator-sdk required by the project if other is on the path
**Describe the bug**
When you have your own version of `operator-sdk`  on the `PATH` it takes precedence over the one defined by the project's `Makefile`. This can lead to the use of incorrect version when generating manifests. 

**Expected behavior**
Use the preferred version of operator-sdk

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

- aligns make target to use the same commands as other tools
- reverts accidental changes in bundle metadata

See https://github.com/opendatahub-io/opendatahub-operator/pull/668#issuecomment-1787416762

Fixes #694

## How Has This Been Tested?
`make bundle`

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -72,7 +72,7 @@ metadata:
     categories: AI/Machine Learning, Big Data
     certified: "False"
     containerImage: quay.io/opendatahub/opendatahub-operator:v2.1.0
-    createdAt: "2023-10-30T14:38:57Z"
+    createdAt: "2023-8-23T00:00:00Z"
```
Comment (zdtsw) [inline bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:75]: ?

File: Makefile [hunk 2]
```diff
@@ -203,27 +203,21 @@ KUSTOMIZE_INSTALL_SCRIPT ?= "https://raw.githubusercontent.com/kubernetes-sigs/k
...
 .PHONY: operator-sdk
-operator-sdk: $(LOCALBIN) ## Download and install operator-sdk into LOCALBIN if not exists in PATH
-ifeq (,$(shell command -v operator-sdk 2>/dev/null))
-ifeq (,$(shell command -v $(OPERATOR_SDK) 2>/dev/null))
-	@{ \
+operator-sdk: $(OPERATOR_SDK) ## Download and install operator-sdk
+$(OPERATOR_SDK): $(LOCALBIN)
 	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
-	curl -sSLo $(OPERATOR_SDK) $(OPERATOR_SDK_DL_URL)/operator-sdk_$${OS}_$${ARCH} && \
+	test -s $(OPERATOR_SDK) || curl -sSLo $(OPERATOR_SDK) $(OPERATOR_SDK_DL_URL)/operator-sdk_$${OS}_$${ARCH} && \
```
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply Makefile:218]: Yes, that was intended I believe

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (bartoszmajsak) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:75]: Don't ask... It's what operator SDK produces

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (zdtsw) [reply Makefile:218]: say, now i want to uplift from 1.24.1 to 1.100.1000, so i modify makefile this OPERATOR_SDK_VERSION.
then, i will need to delete the one from $(OPERATOR_SDK) first, then run "make"

[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply Makefile:218]: Correct. Even without this PR that's what you would need to do anyway. Moreover, this behavior applies to all the other tools, e.g. `kustomize` and `controller-gen`.

Perhaps what we miss in the first place is `make clean` which you can apply after the version bump. You don't have `bin/` in under git control though, so in other, freshly checked-out environments that should be fine.

If we want to go fancy with the `Makefile` we can extend those tool targets to check if the version of locally stored binary matches what's defined in the variable. Not quite sure if that's not going too far with gold-plating the build script :)

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (bartoszmajsak) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:75]: It's essentially the same as in the `main` at this moment: https://github.com/opendatahub-io/opendatahub-operator/blob/6acf1a9c241b642d23a6cc4d9aeb3f8d6e12be35/bundle/manifests/opendatahub-operator.clusterserviceversion.yaml#L75

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (zdtsw) [reply Makefile:218]: i agree. we can get this one out, and see what others can be improved.

**Review by zdtsw — APPROVED** [review]
/lgtm

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (bartoszmajsak) [reply Makefile:218]: I started collecting some ideas here https://github.com/opendatahub-io/opendatahub-operator/issues/696

File: bundle/manifests/opendatahub-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (ykaliuta) [reply bundle/manifests/opendatahub-operator.clusterserviceversion.yaml:75]: Do you have to update it in the patch? I doubt.

**Pull request merged**